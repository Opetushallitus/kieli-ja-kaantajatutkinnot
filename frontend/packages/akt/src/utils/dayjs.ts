import dayjs from 'dayjs';

import { getCurrentLang } from 'configs/i18n';

export const getDayjs = () => {
  dayjs.locale(getCurrentLang());

  return dayjs;
};
