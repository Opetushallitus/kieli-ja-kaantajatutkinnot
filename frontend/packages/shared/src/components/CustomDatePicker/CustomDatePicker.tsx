import { CustomTextFieldProps, CustomTextField } from '../../components';
import { DesktopDatePicker, LocalizationProvider } from '@mui/x-date-pickers';
import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
import dayjs, { Dayjs } from 'dayjs';

interface DatePickerProps extends CustomTextFieldProps {
  value: Dayjs | null;
  setValue: (value: Dayjs | null) => void;
  label?: string;
  disabled?: boolean;
  minDate?: Dayjs;
  maxDate?: Dayjs;
  showToolbar?: boolean;
}

export const CustomDatePicker = ({
  value,
  setValue,
  label = '',
  disabled = false,
  minDate,
  maxDate,
  showToolbar = true,
  error,
  helperText,
  showHelperText,
}: DatePickerProps): JSX.Element => {
  const MIN_DATE = '1980-01-01';
  const MAX_DATE = '2050-01-01';

  const handleChange = (newValue: Dayjs | null) => {
    setValue(newValue);
  };

  return (
    <LocalizationProvider
      dateAdapter={AdapterDayjs}
      adapterLocale={dayjs.locale()}
    >
      <DesktopDatePicker
        className="custom-date-picker"
        showToolbar={showToolbar}
        label={label}
        value={value}
        onChange={handleChange}
        disabled={disabled}
        minDate={minDate ?? dayjs(MIN_DATE)}
        maxDate={maxDate ?? dayjs(MAX_DATE)}
        renderInput={(params) => {
          const textFieldParams = {
            ...params,
            error,
            helperText,
            showHelperText,
          };

          return <CustomTextField {...textFieldParams} />;
        }}
      />
    </LocalizationProvider>
  );
};
