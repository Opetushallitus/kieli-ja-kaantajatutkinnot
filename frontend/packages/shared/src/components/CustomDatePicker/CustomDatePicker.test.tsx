import dayjs from 'dayjs';
import renderer from 'react-test-renderer';

import { CustomDatePicker } from './CustomDatePicker';

describe('DatePicker', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <CustomDatePicker
          label="test label"
          value={dayjs('2022-04-21')}
          setValue={jest.fn()}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
