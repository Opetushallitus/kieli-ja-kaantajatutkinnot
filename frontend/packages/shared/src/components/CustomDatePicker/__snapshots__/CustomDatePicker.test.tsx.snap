// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker should render correctly 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root custom-date-picker css-z3c6am-MuiFormControl-root-MuiTextField-root"
  onClick={[Function]}
  onMouseUp={[Function]}
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink={true}
    htmlFor=":r0:"
    id=":r0:-label"
  >
    test label
  </label>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoComplete="off"
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
      disabled={false}
      id=":r0:"
      inputMode="text"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onPaste={[Function]}
      placeholder="MM/DD/YYYY"
      readOnly={false}
      required={false}
      type="text"
      value="04/21/2022"
    />
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
    >
      <button
        aria-label="Choose date, selected date is Apr 21, 2022"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="CalendarIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
          />
        </svg>
      </button>
    </div>
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-14lo706"
      >
        <span>
          test label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
